auth/user_mgt.go

```diff
@@ -468,15 +522,21 @@ func (c *Client) createUser(ctx context.Context, user *UserToCreate) (string, er
 		user = &UserToCreate{}
 	}
 
-	payload, err := user.preparePayload()
+	request := &identitytoolkit.IdentitytoolkitRelyingpartySignupNewUserRequest{}
+
+	_, err := user.preparePayload(request)
 	if err != nil {
 		return "", err
 	}
-	var rur responseUserRecord
-	if err := c.makeHTTPCall(ctx, "signupNewUser", payload, &rur); err != nil {
+
+	relyingpartySignupNewUserCall := c.is.Relyingparty.SignupNewUser(request)
+	relyingpartySignupNewUserCall.Header().Set("X-Client-Version", c.version)
```

Comment:
I don't like the fact that the header is set in 5 different places. 
(I.e. if we ever change the header, or want to add new identifiers to every call, we'd like to do it in one place)
I don't see a super elegant way to fix this though.

maybe something along the lines of:
(this isn't ideal, ideally I'd also refactor in the next line (and add "Do() (interface{}, error)", and "Context(ctx) *\<interfaceName\>" to the new interface), but returning an interface{} from Do() and casting it later seems ugly.)
```
// Create a new interface
type headerable interface {
    Header() http.Header
}

// refactor this lines 
func (c *Client)execute(ic *headerable) (){
    ic.Header().Set("X-Client-Version", c.version)
}
```
WDYT? do you have another solution?

Reply:
I think it's a good compromise, i was looking for a better solution, but i didn't find anything more elegant :(